rtos/rtos_idle.c

```diff
@@ -22,14 +22,50 @@
 
 #include "rtos/rtos_idle.h"
 #include "platform/mbed_sleep.h"
+#include "mbed_critical.h"
+#include "hal/ticker_api.h"
+#include "hal/lp_ticker_api.h"
+#include "cmsis_os2.h"
+#include "rtx_lib.h"
+#include <limits.h>
+
+static ticker_event_t idle_loop_event;
+
+void idle_loop_handler(uint32_t id)
+{
+    (void)id;
+}
 
 static void default_idle_hook(void)
 {
-    /* Sleep: ideally, we should put the chip to sleep.
-       Unfortunately, this usually requires disconnecting the interface chip (debugger).
-       This can be done, but it would break the local file system.
-    */
+#if DEVICE_LOWPOWERTIMER
+    timestamp_t time_in_sleep = 0UL;
+
+    core_util_critical_section_enter();
+    uint32_t tick_freq = svcRtxKernelGetTickFreq();
```

Comment:
If the critical section is removed and `svcRtxKernelSuspend` is replaced by `osKernelSuspend` and a thread is set as ready and run between the `osKernelSuspend` and `sleep` then the number of ticks acquired by `osKernelSuspend` is invalid and the OS might be put to sleep for an incorrect amount of time.